CRM/Contact/BAO/Contact/Utils.php

```diff
@@ -55,11 +55,7 @@ public static function getImage($contactType, $urlOnly = FALSE, $contactId = NUL
         $imageInfo[$contactType]['url'] = $imageUrl;
       }
       else {
-        $isSubtype = (array_key_exists('parent_id', $typeInfo) &&
-          $typeInfo['parent_id']
-        ) ? TRUE : FALSE;
-
-        if ($isSubtype) {
+        if (!empty($typeInfo['parent_id'])) {
```

Comment:
This one takes the prize for overly complicated. 5 lines to check if a value isset, obtusely cast it to a boolean, assign it to a variable (which is only used once) and then read that variable.